seclang/parser_test.go

```diff
@@ -40,6 +40,19 @@ func TestDirectivesCaseInsensitive(t *testing.T) {
 	}
 }
 
+func TestCommentsWithBackticks(t *testing.T) {
+	waf := coraza.NewWAF()
+	p := NewParser(waf)
+	err := p.FromString(
+		"# This comment has a trailing backtick:`" +
+			`
+		SecAction "id:1,deny,log,phase:1"
```

Comment:
Is this a real use of backticks? Would be good to add some real world test cases, can have multiple with a table test

Reply:
Ah I see that the issue is about using backticks outside of backtick use cases. We can probably make this still a bit more realistic to make it easier to grok

```
(
"# This comment has a trailing backtick `here`" + `
SecAction "id:1,deny,log,phase:1"
`)
```

Also note moving the backtick from line 48 to 47, it looked like part of the string at first, think that makes it easier

Reply:
Thanks Rag, done.
Relas use cases are [CRS comments](https://github.com/coreruleset/coreruleset/blob/v4.0/dev/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf#L1351) like:
- ```# The minimal string that triggers this regexp is: `if` ```
- ``# The rule 942510 is related to 942110 which catches a single ' or ` ``

Reply:
I think the second case is even more interesting for the test rather than the first one as it actually might close the backticks because it holds only one.